app/src/main/java/com/swent/skillswap/MainActivity.kt

```diff
@@ -191,6 +197,39 @@ fun SkillSwapApp(navController: NavHostController = rememberNavController()) {
                         onBackClick = { navigationActions.goBack() }
                     )
                 }
+                composable(Screen.PersonalPosts.route) {
+                    val postRepository = PostFirestoreRepository(Firebase.firestore)
+                    PersonalPostsScreen(
+                        onGoBack = { navigationActions.goBack() },
+                        onEditPost = { post ->
+                            // Navigate to edit screen based on post type
+                            if (post.type == PostType.REQUEST) {
+                                navController.navigate(Screen.EditRequest.createRoute(post.uid))
```

Comment:
The Screen.EditRequest.route uses a path parameter {postId}, but there's no null-safety check when navigating